cpp/dolfinx/mesh/Mesh.cpp

```diff
@@ -112,8 +117,19 @@ Mesh mesh::create(MPI_Comm comm,
     }
   }
 
+  // FIXME: improve. Clip cell data, removing any unused ghost cells
+  // before sending to create_geometry. Maybe AdjacencyList can have a "resize"
+  // method?
+  int n_cells_local = topology.index_map(tdim)->size_local()
+                      + topology.index_map(tdim)->num_ghosts();
+  Eigen::Matrix<std::int32_t, Eigen::Dynamic, 1> off1
+      = cell_nodes.offsets().head(n_cells_local + 1);
+  Eigen::Matrix<std::int64_t, Eigen::Dynamic, 1> data1
+      = cell_nodes.array().head(off1[n_cells_local]);
+  graph::AdjacencyList<std::int64_t> cell_nodes_2(data1, off1);
```

Comment:
Use 
```cpp
graph::AdjacencyList<std::int64_t> cell_nodes_2(cell_nodes.offsets().head(n_cells_local + 1), 
                                                cell_nodes.array().head(off1[n_cells_local]));
```

Reply:
I can't get that to work - some kind of type inference problems, but a const reference seems OK.

Reply:
ok, we can look at a solution later.